app/models/document.rb

```diff
@@ -73,6 +73,7 @@ def main
       {
         heading: {
           text: heading.text,
+          id: heading[:id],
```

Comment:
I'm testing the links on this page

`/how-to-publish-on-gov-uk/creating-and-updating-pages`

There is a link to `#attachments` however the actual ID is `adding-attachments`

From viewing the workflow I'm thinking the IDs are done manually vs programmatically. Investigating a little more

Reply:
If you test with the latest version of the gem PR this should be working as expected